numba/cuda/tests/cudapy/test_atomics.py

```diff
@@ -530,6 +569,111 @@ def kernel(x):
 
         self._test_atomic_returns_old(kernel, np.nan)
 
+    # Tests for atomic nanmin/nanmax
+
+    # nanmax tests
+    def check_atomic_nanmax(self, dtype, lo, hi):
+        vals = np.random.randint(lo, hi, size=(32, 32)).astype(dtype)
+        vals[1::2] = np.nan
+        res = np.zeros(1, dtype=vals.dtype)
+        cuda_func = cuda.jit(atomic_nanmax)
+        cuda_func[32, 32](res, vals)
+        gold = np.nanmax(vals)
+        np.testing.assert_equal(res, gold)
+
+    def test_atomic_nanmax_int32(self):
+        self.check_atomic_nanmax(dtype=np.int32, lo=-65535, hi=65535)
+
+    def test_atomic_nanmax_uint32(self):
+        self.check_atomic_nanmax(dtype=np.uint32, lo=0, hi=65535)
+
+    @skip_unless_cc_32
+    def test_atomic_nanmax_int64(self):
+        self.check_atomic_nanmax(dtype=np.int64, lo=-65535, hi=65535)
+
+    @skip_unless_cc_32
+    def test_atomic_nanmax_uint64(self):
+        self.check_atomic_nanmax(dtype=np.uint64, lo=0, hi=65535)
+
+    def test_atomic_nanmax_float32(self):
+        self.check_atomic_nanmax(dtype=np.float32, lo=-65535, hi=65535)
+
+    def test_atomic_nanmax_double(self):
+        self.check_atomic_nanmax(dtype=np.float64, lo=-65535, hi=65535)
+
+    def test_atomic_nanmax_double_shared(self):
+        vals = np.random.randint(0, 32, size=32).astype(np.float64)
+        vals[1::2] = np.nan
+        res = np.array([0], dtype=vals.dtype)
+        cuda_func = cuda.jit('void(float64[:], float64[:])')(atomic_nanmax_double_shared)
```

Comment:
long lines. maybe good to un-ban this file from flake8

Reply:
unbanned and fixed in 1b51ebc